### DISCUSSION TITLE
Mutliple VNET CIDR's && retro fit to existing VNET's

### DISCUSSION THREAD

**Discussion by georgegil** [Azure/ipam#325] [Q&A] [answered]
So far, glad I found this, this will change the way we do our VNET's significantly. 

## 1) retro fit existing VNETS
we're a Terraform shop and have currently over 150 VNETs deploy with our trusty module. 

we decided to have the `region` ==  `space`

but a space can contain multiple blocks. 

for example eastus2 has two /15 cidr blocks.

to create a new reservation for a new vnet is easy, just `shuffle` on the blocks inside the json and post to the `/api/spaces/eastus2/reservations` API, I get a cidr back with a reservation ID to tag the vnet with... easy peasy.

but if I am to retrofit this to an existing Terraform deployed VNET, I can't predict which block will be assigned, and I can't seem to find a way of searching the block by CIDR.. unless I am missing something?

## 2) multi CIDR vnets:

Most of our VNET's we start with a /24 CIDR, but over time this may grow, at which point we'll add a second CIDR. With IPAM, do I just add multpole X-IPAM-RES-ID tags?

**Comment by DCMattyG** [comment, marked as answer]
Hi @georgegil, thanks for the positive feedback and hopefully I can answer your questions....

For item 1, the **Block** level API will allow you to specify a CIDR when creating a Reservation.

<img width="713" alt="image" src="https://github.com/user-attachments/assets/fa7fb6ee-a2b0-423a-a309-c24851b5cfeb">

For item 2, you can provide a single `X-IPAM-RES-ID` tag with multiple, comma-separated values (ID's).

I hope that helps. Please let us know if you need anything else!

**Reply by georgegil** [reply]
thank you for the quick answer! on a Sunday no less!

re question 2, grand! happy with that. 

re question 1, my issue is that I don't know the block upfront, a CIDR could have been plucked from block `x` or `y` so do not know how to structure my api. 

That being said, I've just figured it out with the `api/tools/cidrCheck` API, which can give me metadata on an existing CIDR..

reckon we're good now!
